server/controllers/UsersController.js

```diff
@@ -0,0 +1,241 @@
+import Helpers from '../utils/helper';
+import { User } from '../models';
+
+/**
+ * @class UsersController
+ */
+class UsersController {
+  /**
+   * @description
+   * @param {any} req
+   * @param {any} res
+   * @returns {void}
+   * @memberof UsersController
+   */
+  static registerUser(req, res) {
+    User.find({
+      where: {
+        email: req.body.email
+      }
+    })
+      .then((user) => {
+        if (!user) {
+          return Helpers.createUser(req, res);
+        }
+        return res.status(401).send({
+          success: false,
+          message: 'This user already exists!'
+        });
+      })
+      .catch(err => res.status(401).send(err));
+  }
+
+  /**
+   * @description
+   * @static
+   * @param {any} req
+   * @param {any} res
+   * @returns {void}
+   * @memberof UsersController
+   */
+  static loginUser(req, res) {
+    User.find({
+      where: {
+        email: req.body.email
+      }
+    })
+      .then((returningUser) => {
+        if (!returningUser) {
+          return res.status(401).send({
+            success: false,
+            message: 'Sorry, the email does not exist!'
+          });
+        }
+        const user = new User();
+        const checkPassword = user.validatePassword(
+          req.body.password,
+          returningUser.password
+        );
+        if (!checkPassword) {
+          return res.status(401).send({
+            message: 'Invalid password'
+          });
+        }
+        const token = user.generateJWT(returningUser.id, returningUser.role);
+        return res.status(200).send({
+          success: true,
+          message: 'Login successful! :)',
+          token,
+          userDetails: {
+            name: returningUser.fullName,
+            id: returningUser.id,
+            role: returningUser.role
+          }
+        });
+      })
+      .catch(err => res.status(401).send(err));
+  }
+
+  /**
+   * @description
+   * @static
+   * @param {any} req
+   * @param {any} res
+   * @returns {object} response
+   * @memberof UsersController
+   */
+  static createUser(req, res) {
+    if (req.headers.authorization) {
+      if (req.decoded.role === 'user') {
+        return res.status(401).send({
+          success: false,
+          message: 'Unathorized access! Only an Admin can create a user'
+        });
+      }
+      return UsersController.registerUser(req, res);
+    }
+    return res.status(400).send({
+      message: 'Please set token in the header'
+    });
+  }
+
+  /**
+   * @description
+   * @static
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} response
+   * @memberof UsersController
+   */
+  static getUsers(req, res) {
+    if (req.decoded.role !== 'admin') {
```

Comment:
Having an `isAdmin()` middleware will help to prevent repeated code in the controller that checks if a user is an admin or not.